activate

```diff
@@ -24,46 +24,50 @@
 #  serial (from DMI / Endless mfgdata)
 #  live (true if this is a live USB, false if not)
 
-set -e
-
-DIR=/var/lib/eos-phone-home
-
-if [ ! -d $DIR -o ! -w $DIR ]; then
+DIR="/var/lib/eos-phone-home"
+if [ ! -d "${DIR}" ] || [ ! -w "${DIR}" ]; then
+    echo "${DIR} missing or not writable"
     exit 1
 fi
 
-ACTIVATEDFILE=$DIR/activated
-
 # Give me a ping, Vasili. One ping only, please.
-if [ -e $ACTIVATEDFILE ]; then
+ACTIVATEDFILE="${DIR}/activated"
+if [ -e "${ACTIVATEDFILE}" ]; then
+    echo "${ACTIVATEDFILE} exists, doing nothing"
     exit 0
 fi
 
-SCRIPT=https://home.endlessm.com/v1/activate
+if [ ! -z "$EOS_PHONE_HOME_ACTIVATE_URL" ]; then
+    SCRIPT="${EOS_PHONE_HOME_ACTIVATE_URL}"
+else
+    SCRIPT="https://home.endlessm.com/v1/activate"
+fi
 
-image=$(attr -q -g eos-image-version /sysroot) || image=''
+image=$(attr -q -g eos-image-version /sysroot) || true
+if [ ! -z "${image}" ]; then
```

Comment:
You use these a lot so maybe doing `if [ "${VAR}" != "" ]` would be better for all of these